### PR TITLE
[admin]Ticket system port

### PR DESCRIPTION
**Pull request by Kmc2000**
This is still a work in progress, but I need a few things from you:

Testing with multiple ckeys
Let me know if any ui glitches occur
Tell me any stylistic changes you want from them.

Ticket panel:

I need to fix their formatting etc. as well as merging open, resolved, and closed tickets into one ticket panel, but please do mess around with them and try to break them!

Todo:
Improve the look with boxes around certain things
merge open, resolved, and closed tickets into one ticket panel
Fix the formatting up a bit.

Hopefully these are less disgusting than TG's current ones.

I'm going for the best of both worlds option and keeping the neat coding of TG's tickets with our cool style of tickets (to also avoid changing 60 odd files).

Feedback needed

(told you I'd have em by sundown :^) )

#### Changelog

:cl: Kmc2000
rscadd: Tickets are now more similar to the classic yog tickets
/:cl:

### CODE REVIEW

**Comment by DLCabbose** [comment]
Nice work, comrade.

**Comment by Moja10** [comment]
Good work! I like it a lot and my only gripe is the purple text, other than that, i'm fine with this getting merged

**Comment by Wardog390** [comment]
what is REJT

**Comment by Kmc2000** [comment]
Reject - Aka the oakboscage of ticket responses.

Just flat out denies the ticket without resolving.

**Comment by Wardog390** [comment]
i like it

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -79,13 +80,23 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 			title = "Resolved Tickets"
 	if(!l2b)
 		return
-	var/list/dat = list("<html><head><title>[title]</title></head>")
-	dat += "<A href='?_src_=holder;[HrefToken()];ahelp_tickets=[state]'>Refresh</A><br><br>"
+//	var/list/dat = list("<html><head><title>[title]</title></head>")
+	var/list/dat = list(get_html("Admin Tickets", "", ""))
+	var/total
+	if(resolved_tickets.len == 0)
```
Comment (Amelia0010) [inline code/modules/admin/verbs/adminhelp.dm:86]: if(!resolved_tickets.len)

File: code/modules/admin/verbs/adminhelp.dm [hunk 2]
```diff
@@ -222,21 +233,24 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
-	. = " (<A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=reject'>REJT</A>)"
-	. += " (<A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=icissue'>IC</A>)"
-	. += " (<A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=close'>CLOSE</A>)"
-	. += " (<A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=resolve'>RSLVE</A>)"
+	. = " <A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=reject'>REJT</A>" //The brackets looed ugly as shit
+	. += " <A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=icissue'>IC</A>"
+	. += " <A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=close'>CLOSE</A>"
+	. += " <A HREF='?_src_=holder;[HrefToken(TRUE)];ahelp=[ref_src];ahelp_action=resolve'>RSLVE</A>"
+
+//"[ADMIN_QUE(user)] [ADMIN_PP(user)] [ADMIN_VV(user)] [ADMIN_SM(user)] [ADMIN_FLW(user)] [ADMIN_TP(user)] [ADMIN_INDIVIDUALLOG(user)] [ADMIN_SMITE(user)]"
```
Comment (Amelia0010) [inline code/modules/admin/verbs/adminhelp.dm:252]: rid this

File: code/modules/admin/verbs/adminhelp.dm [hunk 3]
```diff
@@ -384,35 +398,34 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
+//	var/list/dat = list("<html><head><title>Ticket #[id]</title></head>")
+	var/list/dat = list(get_html("Ticket Log Viewer", "", ""))
 	var/ref_src = "[REF(src)]"
-	dat += "<h4>Admin Help Ticket #[id]: [LinkedReplyName(ref_src)]</h4>"
-	dat += "<b>State: "
+	dat += "<h4>Ticket#[id]</h4>"
+	dat += "[LinkedReplyName(ref_src)]"
+//	dat += "<br><br>Opened at: [gameTimestamp(wtime = opened_at)] (Approx [DisplayTimeText(world.time - opened_at)] ago)"  These are clutter, IMO
+//	if(closed_at)
+//		dat += "<br>Closed at: [gameTimestamp(wtime = closed_at)] (Approx [DisplayTimeText(world.time - closed_at)] ago)"
```
Comment (Amelia0010) [inline code/modules/admin/verbs/adminhelp.dm:408]: and this

File: code/modules/admin/verbs/ahelp_template.dm [hunk 4]
```diff
@@ -0,0 +1,238 @@
+/proc/get_html(title as text, js as text, css as text, content as text)
```
Comment (Amelia0010) [inline code/modules/admin/verbs/ahelp_template.dm:1]: this whole thing should be in the modular folder

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
now it just lacks the //yogs bits

File: code/modules/admin/verbs/ahelp_template.dm (hunk 4 above)
Comment (Kmc2000) [reply code/modules/admin/verbs/ahelp_template.dm:1]: Didn't know we had one

Comment (Amelia0010) [reply code/modules/admin/verbs/ahelp_template.dm:1]: it's the yogstation/yogstation folder 😅

**Comment by DLCabbose** [comment]
So where is this now in terms of being ready for rebase now?

**Comment by Kmc2000** [comment]
I'm working on gimmicky stuff for my server.
The rest of this is complicated HTML.

They are ready for testing but they don't look perfect yet.

File: code/modules/admin/verbs/adminhelp.dm (hunk 1 above)
Comment (Moja10) [reply code/modules/admin/verbs/adminhelp.dm:86]: +1

**Pull request closed without merging**